### PR TITLE
Makefile improvements

### PR DESCRIPTION
**Pull request by slimreaper35**
# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] Docs updated (if applicable)
- [ ] Docs links in the code are still valid (if docs were updated)

**Note:** if the contribution is external (not from an organization member), the CI
pipeline will not run automatically. After verifying that the CI is safe to run:

- [approve GitHub Actions workflows][approve-gh-actions] by clicking a button
- approve the Red Hat Trusted App Pipeline container build by commenting `/ok-to-test`
  (as is the [standard for Pipelines as Code][pac-running-pipeline])

[approve-gh-actions]: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
[pac-running-pipeline]: https://pipelinesascode.com/docs/guide/running/#running-the-pipelinerun

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,13 +17,10 @@ test: venv
 	venv/bin/tox
 
 test-unit: venv
-	venv/bin/tox -e $(TOX_ENVLIST) -- $(TOX_ARGS)
+	venv/bin/tox -e py3
```
Comment (brunoapimentel) [inline Makefile:20]: I don't mind this change, but I find useful to have the TOX_ARGS variable here, as it allows us to filter which tests to execute:
```
TOX_ARGS="-k pip" make test-unit
```

I always missed it in the `test-integration` command, btw.

Comment (brunoapimentel) [reply Makefile:20]: On the other hand, I think I've been mostly calling tox directly, so if nobody else finds it uself, let's drop it.

File: cachi2/core/scm.py [hunk 2]
```diff
@@ -42,7 +43,15 @@ def get_repo_id(repo: Union[str, PathLike[str], Repo]) -> RepoID:
...
-        repo = Repo(repo, search_parent_directories=True)
+        try:
+            repo = Repo(repo, search_parent_directories=True)
+        except (InvalidGitRepositoryError, NoSuchPathError):
+            raise UnsupportedFeature(
+                "Cachi2 cannot process provided path as a git repository",
+                solution=(
+                    "Please ensure that the path is correct and that it is a valid git repository."
+                ),
+            )
```
Comment (eskultety) [inline cachi2/core/scm.py:54]: This commit is irrelevant to the Makefile changes topic, please propose this in a separate PR for cleaner bookkeeping.

File: Makefile (hunk 1 above)
Comment (slimreaper35) [reply Makefile:20]: Personally, I use `tox -e py3 -- -k pip`.

**Review by eskultety — COMMENTED** [review]
What I think we're really doing a bad job here is the non-transparency of the Makefile that removal of variables just isn't going to improve much/at all. I think we're long overdue on something like the following that everyone would benefit from as suddenly people would know what the Makefile is doing:

```
diff --git a/Makefile b/Makefile
index 99224701..39c6f5f3 100644
--- a/Makefile
+++ b/Makefile
@@ -1,48 +1,67 @@
 PYTHON_VERSION_VENV ?= python3
 TOX_ENVLIST ?= py3
 TOX_ARGS ?=
-GENERATE_TEST_DATA = false
-TEST_LOCAL_PYPISERVER = false
-TEST_LOCAL_DNF_SERVER = false
+GENERATE_TEST_DATA ?= false
+TEST_LOCAL_PYPISERVER ?= false
+TEST_LOCAL_DNF_SERVER ?= false
 
-.PHONY: clean pip-compile
-all: venv
+_CONFIG_VARS = \
+       "PYTHON_VERSION_VENV:Python 3 binary name to create a new venv (default: $(PYTHON_VERSION_VENV))" \
+       "TOX_ENVLIST:Override the list of test environments run by tox (default: $(TOX_ENVLIST))" \
+       "TOX_ARGS:Extra tox args passed through after -- (default: $(TOX_ARGS))"
       <<<<< Need the rest of the variables here >>>>
 
-clean:
+.PHONY: clean pip-compile help
+.DEFAULT_GOAL := help
+
+help:  ## Display help
+       @echo "Usage: make [VARIABLE=value] [TARGET]"
+       @echo
+       @echo "Available targets:"
+       @grep -E '^[a-zA-Z_-]+:.*$$' $(MAKEFILE_LIST) | sed 's/:.*## /|/' | while IFS='|' read -r target desc; do \
+               printf "  %-15s\t%s\n" "$$target" "$${desc:-"(no description)"}"; done
+       @echo
+       @echo "Configuration variables:"
+       @for var in $(_CONFIG_VARS); do \
+               VAR_NAME=$${var%%:*}; \
+               VAR_DESC=$${var#*:}; \
+               printf "  %-15s\t%s\n" $$VAR_NAME "$$VAR_DESC"; \
+       done
+
+clean:  ## Clean the project directory
        rm -rf dist venv .tox *.egg-info *.log*
 
-venv:
+venv:  ## Create a complete development virtual env [vars: PYTHON_VERSION_VENV]
        /usr/bin/env $(PYTHON_VERSION_VENV) -m venv venv
        venv/bin/pip install --upgrade pip
        venv/bin/pip install -r requirements.txt -r requirements-extras.txt
        venv/bin/pip install tox
        venv/bin/pip install -e .
 
-test: venv
+test: venv   ## Run the default tox test environment
        venv/bin/tox
 
-test-unit: venv
+test-unit: venv  ## Run unit tests with tox [vars: TOX_ENVLIST, TOX_ARGS]
        venv/bin/tox -e $(TOX_ENVLIST) -- $(TOX_ARGS)
 
-test-integration: venv
+test-integration: venv  ## Run integration tests with tox [vars: TOX_ARGS, GENERATE_TEST_DATA, TEST_LOCAL_PYPISERVER, TEST_LOCAL_DNF_SERVER]
        CACHI2_GENERATE_TEST_DATA=$(GENERATE_TEST_DATA) \
        CACHI2_TEST_LOCAL_PYPISERVER=$(TEST_LOCAL_PYPISERVER) \
                venv/bin/tox -e integration -- $(TOX_ARGS)
        CACHI2_TEST_LOCAL_DNF_SERVER=$(TEST_LOCAL_DNF_SERVER)
 
-mock-unittest-data:
+mock-unittest-data:  ## Refresh unittest package manager mock data
        hack/mock-unittest-data/gomod.sh
 
-build-image:
+build-image:  ## Build a fresh cachi2 image
        podman build -t localhost/cachi2:latest .
 
 # If you're worried that your local image may be outdated
 # (old base image, old rpms cached in the microdnf install layer)
-build-pristine-image:
+build-pristine-image:  ## Build a fresh cachi2 image refreshing its parent dependencies as well
        podman build --pull-always --no-cache -t localhost/cachi2:latest .
 
 # we need git installed in the image due to setuptools-scm which has it as a direct dependency
-pip-compile:
+pip-compile:  ## Refresh our locked dependencies
        @podman run \
        --rm \
        --volume ${PWD}:/cachi2:rw,Z \
```

**Comment by a-ovchinnikov** [comment]
https://github.com/containerbuildsystem/cachi2/issues/662 makes this moot . Let's switch to nox (or tox).

**Pull request closed without merging**